packages/venia-ui/lib/components/SavedPaymentsPage/savedPaymentsPage.js

```diff
@@ -19,54 +18,32 @@ const SavedPaymentsPage = props => {
         }
     });
 
-    const { handleAddPayment, savedPayments } = talonProps;
+    const { savedPayments } = talonProps;
 
     const classes = mergeClasses(defaultClasses, props.classes);
 
-    const savedPaymentElements = useMemo(() => {
-        return savedPayments.map(
-            ({ details, public_hash, payment_method_code }) => (
-                // TODO: Clean up in PWA-636
-                <div key={public_hash}>
-                    <div>{payment_method_code}</div>
-                    <div>{JSON.stringify(details, null, 2)}</div>
-                </div>
-            )
-        );
-    }, [savedPayments]);
+    const savedPaymentElements = useMemo(
+        () =>
+            savedPayments.map(paymentDetails => (
+                <PaymentCard
+                    key={paymentDetails.public_hash}
+                    {...paymentDetails}
```

Comment:
We need to be explicit about the interface for this target; I would use a named prop here.

Reply:
Can you elaborate on this? Sorry, I dint get your point.

Reply:
I initially was thinking this wasn't very extensible if someone changed the query to return other fields, thus changing the interface for this component. Expected to see `data={paymentDetails}` here but seemed like overkill after seeing how simple the response was, and that any extra data would be serialized in `details` anyway.

I instead just added propTypes for the expected query response [here](https://github.com/magento/pwa-studio/pull/2882/commits/d3ea89e0455481f40e05356f2b3dfadb14ebd6f0#diff-f6533ffec4057589b73942585c5f19e161a2fc4353dde8a1c0d70565c1275a3dR14-R17).